### PR TITLE
Fix index.html for compat before 3.9 and dynamic index

### LINKED ISSUE #39909: [3.2] [3.8] Empty welcome page on new projects for old versions of Quarkus
### Describe the bug

When generated apps are downloaded from code.quarkus.io and run in dev mode, there is no welcome page("index.html" with "Congratualtions, you use Quarkus") on localhost.

### Expected behavior

There should be a welcome page, just like for 3.9 and apps, downloaded from code.quarkus.redhat.com

### Actual behavior

Http 404

### How to Reproduce?

1. Download an app: `https://code.quarkus.io/api/download?e=quarkus-resteasy-reactive&S=3.2&j=17`
2. Unpack the zip
3. Run `mvn quarkus:dev`
4. Open localhost:8080 in a browser

If one downloads the latest version (eg `https://code.quarkus.io/api/download?e=quarkus-resteasy-reactive&S=3.9&j=17`) or uses RHBQ (`https://code.quarkus.redhat.com/api/download?e=quarkus-resteasy-reactive&S=3.2&j=17`) the welcome page is there

### Output of `uname -a` or `ver`

6.7.3-200.fc39.x86_64

### Output of `java -version`

Java version: 21.0.1, vendor: Eclipse Adoptium

### Quarkus version or git rev

3.8.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

Probably related to https://github.com/quarkusio/quarkus/pull/37680

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes https://github.com/quarkusio/quarkus/issues/39909 on future CLI (and code.quarkus)

### CODE REVIEW

File: independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/ContentMergeCodestartFileStrategyHandler.java [hunk 1]
```diff
@@ -40,6 +40,10 @@ public void process(Path targetDirectory, String relativePath, List<TargetFile>
             return;
         }
         createDirectories(targetPath);
-        writeFile(targetPath, template.get().getContent().replace("{merged-content}", mergedContent.toString()));
```
Comment (maxandersen) [inline independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/ContentMergeCodestartFileStrategyHandler.java:43]: so basically only write it if it has no contents?

I get that it solves it for now ...but even empty files are worth having.

no other way we can handle it ?

Comment (ia3andy) [reply independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/ContentMergeCodestartFileStrategyHandler.java:43]: Added @deprecated

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 2]
```diff
@@ -0,0 +1,288 @@
+{#if generate-native and quarkus.platform.version.compareVersionTo("3.9") < 0}
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:1]: what does "generate-native" mean?

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:1]: oups that's a bad copy and paste good catch!!!

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html]: fixed!

Comment (ia3andy) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html]: FYI I tested locally and it worked as expected, but that was just luck..

**Review by maxandersen — APPROVED** [review]
LGTM and backported.

**Pull request merged**